src/llmcompressor/pipelines/sequential/helpers.py

```diff
@@ -0,0 +1,365 @@
+
+from compressed_tensors import has_offloaded_params
+from compressed_tensors.quantization import find_name_or_class_matches
+from torch.fx import Graph, GraphModule, Node
+from torch.fx.proxy import Argument
+from torch.nn import Module
+from transformers import PreTrainedModel
+from transformers.configuration_utils import PretrainedConfig
+from transformers.utils.fx import HFTracer
+
+from llmcompressor.modifiers.utils.hooks import HooksMixin
+from llmcompressor.utils.helpers import calibration_forward_context
+
+__all__ = ["trace_subgraphs", "Subgraph"]
+
+
+@dataclass
+class Subgraph:
+    """
+    Dataclass specifying an executable subgraph of a model graph
+
+    :param graph: subgraph of model graph
+    :param input_names: argument names of the compiled forward function
+    :param consumed_names: argument names which are not used by any subsequent subgraphs
+        and can therefore be deleted from the intermediates cache
+    """
+
+    graph: Graph
+    input_names: Set[str]
+    consumed_names: Set[str]
+
+    def compile_forward(self) -> Callable[[Any], Any]:
+        """
+        Generate and compile code for executing this subgraph
+
+        :return: function which, when called, executes this subgraph
+        """
+        code = self.graph.python_code("self")
+        exec(code.src, code.globals)
+        return code.globals.get("forward")
+
+
+def trace_subgraphs(
+    model: PreTrainedModel,
+    sample_input: Dict[str, Any],
+    sequential_targets: List[str],
+    ignore: List[str],
+) -> List[Subgraph]:
+    """
+    Trace a model to produce subgraphs, where each sequential target belongs to exactly
+    one subgraph and where executing each subgraph in order is equivalent to executing
+    the original model
+
+    :param model: model being traced
+    :param sample_input: inputs whose values will change during execution but whose
+        __len__, __bool__, and __contains__ values are assumed constant across batches
+    :param sequential_targets: list of patterns matching sequential targets
+    :param ignore: list of patterns matching modules to ignore during tracing
+    :return: a list of Subgraphs in order of execution
+    """
+    # find modules
+    sequential_targets = match_modules(model, sequential_targets)
+    ignore = match_modules(model, ignore)
+
+    # initialize arguments
+    tracer = get_tracer(model, sequential_targets, ignore)
+    concrete_args = populate_concrete_args(model, sample_input)
+
+    # trace
+    with (
+        calibration_forward_context(model),
+        HooksMixin.disable_hooks(),
+    ):
+        graph = GraphModule(
+            model,
+            tracer.trace(
+                model,
+                dummy_inputs=sample_input,
+                concrete_args=concrete_args,
+                complete_concrete_args_with_inputs_not_in_dummy_inputs=False,
+                # bug in trace throws an error for variadic
+                # args and kwargs in function signature
```

Comment:
Is this just explaining that you couldn't pass `*args, **kwargs` here?

Reply:
This is explaining why I need to write my own `populate_concrete_args` function rather than rely on [the one provided by transformers](https://github.com/huggingface/transformers/blob/main/src/transformers/utils/fx.py#L1268-L1280)